adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java

```diff
@@ -181,22 +229,24 @@ public ResourceIdentifier topic() {
     }
 
     /**
-     * Gets the tenant that the device belongs to that published
-     * the message.
+     * Gets the tenant that the device belongs to that published the message.
      *
-     * @return The tenant identifier or {@code null} if the device is
-     *         not authenticated and the message's topic does not contain
-     *         a tenant identifier.
+     * @return The tenant object.
      */
-    public String tenant() {
+    public TenantObject tenantObject() {
+        return tenantObject;
+    }
 
-        if (authenticatedDevice != null) {
-            return authenticatedDevice.getTenantId();
-        } else if (topic != null) {
-            return topic.getTenantId();
-        } else {
-            return null;
-        }
+    /**
+     * Gets the authentication identifier used in the MQTT connection.
+     * <p>
+     * In case of an unauthenticated connection, the identifier of the device that published the message is returned
+     * here.
```

Comment:
IMHO that is very misleading as we clearly distinguish between authentication identifier and device identifier ...
FMPOV this method should return `null` in that case.

Reply:
I don't really like the inclusion of the device id case here either.
The motivation here was to also enable using the "sampling-mode-per-auth-id" configuration (defined here:
https://github.com/eclipse/hono/blob/3ebcff93588f0294f1df62da1de1147369577cb9/site/documentation/content/api/management/device-registry-v1.yaml#L859-L868
) in the case of using unauthenticated devices, where then simply the device id would be used for that lookup.

I think we could either add a `MqttContext#deviceId` field then and pass along both ids, or, as this case of using the device id in the "sampling-mode-per-auth-id" lookup isn't documented anyway, we could simply remove this feature, also as we consider using unauthenticated requests as not to be the standard scenario.
I tend to prefer doing the latter, WDYT?

Reply:
I would also prefer to simply omit the feature for unauthenticated devices (for the time being) ...

Reply:
Ok, I've removed the feature in this PR for MQTT then.